google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.server.run;
+
+import com.google.cloud.tools.intellij.appengine.cloud.AppEngineExecutor;
+import com.google.cloud.tools.intellij.appengine.cloud.AppEngineRunTask;
+import com.google.cloud.tools.intellij.appengine.server.instance.AppEngineServerModel;
+
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.process.OSProcessHandler;
+import com.intellij.execution.runners.ProgramRunner;
+import com.intellij.javaee.run.configuration.CommonModel;
+import com.intellij.javaee.run.localRun.EnvironmentHelper;
+import com.intellij.javaee.run.localRun.ExecutableObject;
+import com.intellij.javaee.run.localRun.ExecutableObjectStartupPolicy;
+import com.intellij.javaee.run.localRun.ScriptHelper;
+import com.intellij.javaee.run.localRun.ScriptsHelper;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Runs a Google App Engine Standard app locally with devappserver, through the tools lib.
+ */
+public class CloudSdkStartupPolicy implements ExecutableObjectStartupPolicy {
+
+  // The startup process handler is kept so the process can be explicitly terminated, since we're
+  // not delegating that to the framework.
+  OSProcessHandler startupProcessHandler;
+
+  @Nullable
+  @Override
+  public ScriptHelper createStartupScriptHelper(ProgramRunner programRunner) {
+    return new ScriptHelper() {
+      @Nullable
+      @Override
+      public ExecutableObject getDefaultScript(final CommonModel commonModel) {
+        return new ExecutableObject() {
+          @Override
+          public String getDisplayString() {
+            return "App Engine Plugins Core library";
+          }
+
+          @Override
+          public OSProcessHandler createProcessHandler(String string, Map<String, String> map)
+              throws ExecutionException {
+            AppEngineServerModel runConfiguration;
+            try {
+              // Getting the clone so the debug flags aren't added to the persisted settings.
```

Comment:
Ok. I'm sure I understand this (why would they be persisted over and over again..seems like a symptom of something that should be refactored). But if there's no other way I'll take your word for it.